projects/RabbitMQ.Client/client/impl/SessionBase.cs

```diff
@@ -126,13 +127,11 @@ public void Notify()
 
         public virtual void Transmit(in OutgoingCommand cmd)
         {
-            if (CloseReason != null)
+            if (cmd.Method.ProtocolCommandId != client.framing.ProtocolCommandId.ChannelCloseOk && !IsOpen)
```

Comment:
Why swap the commandId first? The common case is that they're not equal and thus have to check !IsOpen anyway.

Reply:
Makes sense. I was going for the cheaper case first (instead of the more common) which doesn't make sense since it'll almost always be hit anyway.